### PR TITLE
TiCDC/Kafka: Change from WithMaxTries to WithTotalRetryDuration

### LINKED ISSUE #9504: TiCDC kafka sink retry is too short.
### What did you do?

Failure on the kafka sink results in fatal error due to too short retry time.

### What did you expect to see?

Proper retry on kafka connection failures

### What did you see instead?

```
[2023/08/04 06:39:28.761 +00:00] [WARN] [admin.go:100] ["query kafka cluster meta failed, retry it"] [namespace=default] [changefeed=foobar-2023-08-03] [error=EOF]
[2023/08/04 06:39:28.796 +00:00] [WARN] [admin.go:100] ["query kafka cluster meta failed, retry it"] [namespace=default] [changefeed=foobar-2023-08-03] [error="write tcp 4.3.2.1:44712->4.3.2.2:9093: write: broken pipe"]
[2023/08/04 06:39:28.897 +00:00] [WARN] [admin.go:100] ["query kafka cluster meta failed, retry it"] [namespace=default] [changefeed=foobar-2023-08-03] [error="write tcp 4.3.2.1:44712->4.3.2.2:9093: write: broken pipe"]
[2023/08/04 06:39:28.897 +00:00] [WARN] [kafka_manager.go:167] ["Kafka admin client describe topics failed"] [namespace=default] [changefeed=foobar-2023-08-03] [error="[CDC:ErrReachMaxTry]reach maximum try: 3, error: write tcp 4.3.2.1:44712->4.3.2.2:9093: write: broken pipe: write tcp 4.3.2.1:44712->4.3.2.2:9093: write: broken pipe"] [errorVerbose="[CDC:ErrReachMaxTry]reach maximum try: 3, error: write tcp 4.3.2.1:44712->4.3.2.2:9093: write: broken pipe: write tcp 4.3.2.1:44712->4.3.2.2:9093: write: broken pipe\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20221009092201-b66cddb77c32/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20221009092201-b66cddb77c32/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/retry.run\n\tgithub.com/pingcap/tiflow/pkg/retry/retry_with_opt.go:69\ngithub.com/pingcap/tiflow/pkg/retry.Do\n\tgithub.com/pingcap/tiflow/pkg/retry/retry_with_opt.go:34\ngithub.com/pingcap/tiflow/pkg/sink/kafka.(*saramaAdminClient).queryClusterWithRetry\n\tgithub.com/pingcap/tiflow/pkg/sink/kafka/admin.go:92\ngithub.com/pingcap/tiflow/pkg/sink/kafka.(*saramaAdminClient).GetTopicsMeta\n\tgithub.com/pingcap/tiflow/pkg/sink/kafka/admin.go:251\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/mq/manager.(*kafkaTopicManager).getMetadataOfTopics\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/mq/manager/kafka_manager.go:165\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/mq/manager.(*kafkaTopicManager).backgroundRefreshMeta\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/mq/manager/kafka_manager.go:106\nruntime.goexit\n\truntime/asm_arm64.s:1172"] [duration=139.603489ms]
```

Looks like `(*saramaAdminClient) queryClusterWithRetry()` is using `retry.WithBackoffBaseDelay()` with a delay/retry of `
`defaultRetryBackoff`/`defaultRetryMaxTries` which are set to 20 and 3. However as the argument for `WithBackoffBaseDelay()` is in Ms this isn't long enough to do a proper backoff.

https://github.com/pingcap/tiflow/blob/a2f5a741e9d11a0e1ce11191ae22265b02de6a67/pkg/sink/kafka/admin.go#L45-L46

https://github.com/pingcap/tiflow/blob/a2f5a741e9d11a0e1ce11191ae22265b02de6a67/pkg/sink/kafka/admin.go#L109-L109

https://github.com/pingcap/tiflow/blob/a2f5a741e9d11a0e1ce11191ae22265b02de6a67/pkg/retry/options.go#L55

The result is a TiCDC changefeed that's dead in the water. 

### Versions of the cluster

TiCDC v7.1.1

This is a possible regression of #8225 and #8223

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #9504 

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
The retry time for the kafka sink has been increased
```

### CODE REVIEW

File: pkg/sink/kafka/admin.go [hunk 1]
```diff
@@ -42,8 +43,8 @@ type saramaAdminClient struct {
 }
 
 const (
-	defaultRetryBackoff  = 20
-	defaultRetryMaxTries = 3
+	defaultRetryBackoff     = 20
+	defaultRetryMaxDuration = time.Duration(120) * time.Second
```
Comment (sdojjy) [inline pkg/sink/kafka/admin.go:47]: Is 120 seconds too long?

Comment (dveeden) [reply pkg/sink/kafka/admin.go:47]: Restarting a network switch or restarting kafka might take some time. Why would retrying for 2m be too long?

**Comment by dveeden** [comment]
Closing this as #9511 fixes this by removing the retry and relying on the sarama lib to do this instead.

**Pull request closed without merging**